### ISSUE TITLE
dotnet new returns wrong exit code when required parameter is not specified with dotnet sdk 7

### ISSUE THREAD

**Issue by Gcaya** [dotnet/sdk#29157]
### Describe the bug
Dotnet sdk 7 returns a different error message than version 6 when a required dotnet template parameter is not specified using the dotnet new command.

### To Reproduce

1. Clone [repository](https://github.com/Gcaya/dotnet-templating)
2. At the root of the repo run  `dotnet new -i cayawebapi`
3. Now a template with short name as **_cayawebapi_** should be visible when running `dotnet new --list`
4. Then using dotnet sdk version 6  (I have 6.0.403) if we run `dotnet new cayawebapi` in any empty folder without specifying the --templateString parameter, which is required, we'll get the error below which makes sense.
![image](https://user-images.githubusercontent.com/8711024/203225596-e753c85b-6e57-4f4e-8171-7da3ae10e622.png)
5. Now if we install dotnet sdk 7 (I'm at 7.0.100 when I run dotnet --version) and redo step 4, the error message is different. A code 103 is returned meaning the template is not found but we haven't uninstalled it.
![image](https://user-images.githubusercontent.com/8711024/203226009-0a8187f0-27bf-49fb-a733-3f2e2850065a.png)
6. If we run `dotnet new cayawebapi --templateString "foo bar"` specifying the required parameter, then it works fine.

Note: It doesn't matter if the template was installed using version 6 or 7, the above behaviour remains the same.

### Expected behaviour
I would expect the error message to be the same from version 6 to 7, saying that a required parameter is missing and not that no template could be found.

 
### Further technical details
`dotnet --info` output:
```
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.100/

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  6.0.403 [/usr/share/dotnet/sdk]
  8.0.100 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found
```

Using Rider Build #RD-222.4167.23
Command were executed inside a Ubuntu terminal on Windows 10

**Comment by MarcoRossignoli** [comment]
cc: @vlada-shubina

**Comment by vlada-shubina** [comment]
@Gcaya 

thank you for reporting it!

It's a duplicate of https://github.com/dotnet/templating/issues/5600 and will be fixed in 7.0.101.
Please follow the available workarounds in the issue above if needed.
